### PR TITLE
Documentation Part V: channeld

### PR DESCRIPTION
**Pull request by rustyrussell**

### CODE REVIEW

File: openingd/openingd.c [hunk 1]
```diff
@@ -877,6 +1040,19 @@ static u8 *fundee_channel(struct state *state, const u8 *open_channel_msg)
...
+		 *...
+		 * - when failure was caused by an invalid signature check:
+		 *    - SHOULD include the raw, hex-encoded transaction in reply
+		 *      to a `funding_created`, `funding_signed`,
+		 *      `closing_signed`, or `commitment_signed` message.
+		 */
+		/*~ This verbosity is not only useful for our own testing, but
+		 * a courtesy to other implementaters whose brains may be so
+		 * twisted by coding in Go, Scala and Rust that they can no
+		 * longer read C code. */
```
Comment (conscott) [inline openingd/openingd.c]: :laughing:

File: openingd/openingd.c [hunk 2]
```diff
@@ -36,23 +46,34 @@
...
 #define GOSSIP_FD 4
 #define HSM_FD 5
 
+/* Global state structure.  This is only for the one specific peer and channel */
 struct state {
+	/* Cryptographic state needed to exchange messages with the peer */
 	struct crypto_state cs;
-	struct pubkey next_per_commit[NUM_SIDES];
 
 	/* Constriants on a channel they open. */
```
Comment (conscott) [inline openingd/openingd.c]: Spelling: Constriants -> Constraints

File: openingd/openingd.c [hunk 3]
```diff
@@ -62,22 +83,37 @@ struct state {
...
+/*~ If we can't agree on parameters, we fail to open the channel.  If we're
+ * the funder, we need to tell lightningd, otherwise it never really notices. */
 static void negotiation_aborted(struct state *state, bool am_funder,
 				const char *why)
 {
-	status_debug("aborted opening negotiaion: %s", why);
+	status_debug("aborted opening negotiation: %s", why);
+	/*~ The "billboard" (exposed as "status" in the JSON listpeers RPC
+	 * call) is a transient per-channel area which indicates important
+	 * information about what is happening.  It has a "permenant" area for
```
Comment (conscott) [inline openingd/openingd.c]: Spelling: permenant -> permanent

**Comment by conscott** [comment]
Loving more docs! Thank you.

File: openingd/openingd.c [hunk 4]
```diff
@@ -1163,20 +1342,25 @@ int main(int argc, char *argv[])
 				   &inner))
 		master_badmsg(WIRE_OPENING_INIT, msg);
 
-	/* If they wanted to send an msg, do so before we waste time
-	 * doing work.  If it's a global error, we'll close
-	 * immediately. */
+	/*~ If they wanted to send an msg, do so before we waste time doing
```
Comment (niftynei) [inline openingd/openingd.c]: `an msg` -> `a msg`

File: openingd/openingd.c [hunk 5]
```diff
@@ -297,26 +368,36 @@ static u8 *opening_negotiate_msg(const tal_t *ctx, struct state *state,
...
 						    err));
 			/* Return NULL so caller knows to stop negotiating. */
 			return NULL;
 		}
 
+		/*~ We do not support multiple "live" channels, though the
+		 * protocol has a "channel_id" field in all non-gossip messages
+		 * so it's possible.  Our one-process-one-channel mechanism
+		 * keeps things simple: if we wanted to change this, we would
+		 * probably be best with another daemon to de-multipliex them;
```
Comment (wythe) [inline openingd/openingd.c]: multipliex -> multiplex

**Comment by niftynei** [comment]
ACK https://github.com/ElementsProject/lightning/pull/2132/commits/a2cf452b11c0330072b96561eb9f49954bb2755e

**Pull request merged**